docs/api/specs/desktopAgentCommunicationProtocol.md

```diff
@@ -0,0 +1,448 @@
+
+- [`privateChannelUnsubscribeEventListenerRequest`](https://fdc3.finos.org/schemas/next/api/privateChannelUnsubscribeEventListenerRequest.schema.json)
+- [`privateChannelUnsubscribeEventListenerResponse`](https://fdc3.finos.org/schemas/next/api/privateChannelUnsubscribeEventListenerResponse.schema.json)
+
+#### `disconnect()`
+
+Request and response used to implement the [`PrivateChannel.disconnect()`](../ref/PrivateChannel#disconnect) API call:
+
+- [`privateChannelDisconnectRequest`](https://fdc3.finos.org/schemas/next/api/privateChannelDisconnectRequest.schema.json)
+- [`privateChannelDisconnectResponse`](https://fdc3.finos.org/schemas/next/api/privateChannelDisconnectResponse.schema.json)
+
+### Checking apps are alive
+
+Depending on the connection over which the Desktop Agent and app are connected, it may be necessary for the Desktop Agent to check whether the application is still alive. This can be done, either periodically or on demand (for example to validate options that will be provided in an [`AppIntent`](../ref/Metadata#appintent) as part of a `findIntentResponse` or `raiseIntentResponse` and displayed in an intent resolver interface), using the following message exchange:
+
+- [`heartbeatEvent`](https://fdc3.finos.org/schemas/next/api/heartbeatEvent.schema.json)
+- [`heartbeatAcknowledgment`](https://fdc3.finos.org/schemas/next/api/heartbeatAcknowledgment.schema.json)
+
+As a Desktop Agent initiated exchange, it is initiated with an `AgentEvent` message and completed via an `AppRequest` message as an acknowledgement.
+
+:::tip
+
+Additional procedures are defined in the [Browser Resident Desktop Agents specification](./browserResidentDesktopAgents#disconnects) and [Web Connection Protocol](./webConnectionProtocol#step-5-disconnection) for the detection of app disconnection or closure. Implementations will often need to make use of multiple procedures to catch all forms of disconnection in a web browser.
+
+:::
+
+### Controlling injected User Interfaces
+
+Desktop Agent implementations, such as those based on the [Browser Resident Desktop Agents specification](./browserResidentDesktopAgents) and [Web Connection Protocol](./webConnectionProtocol), may either provide their own user interfaces (or other appropriate mechanisms) for the selection of User Channels or Intent Resolution, or they may work with implementations injected into the application (for example, as described in the [Web Connection Protocol](./webConnectionProtocol#providing-channel-selector-and-intent-resolver-uis) and implemented in [`getAgent()`](../ref/GetAgent)).
+
+Where injected user interfaces are used, standardized messaging is needed to communicate with those interfaces. This is provided in the DACP via the following 'iframe' messages, which are governed by the [`iFrameMessage`](https://fdc3.finos.org/schemas/next/api/iFrameMessage.schema.json) schema. The following messages are provided:
```

Comment:
is iFrame supposed to be in single quotes or backticks? I don't think we use single quotes anywhere else.

Reply:
In this case yes - the class of message referred to will be used for controlling the iframes in fdc3 for the web/getAgent() but _might_ also be used to control other UIs in other languages. However, there are web-centric properties to them so that might not be appropriate...

To be honest, I think they should be part of the WCP rather than DACP due to the web-centric nature, hello/handshake pattern recycled from the WCP etc.. I believe @robmoffat is of the opposite opinion? Which they belong to doesn't really make much difference other than the file they are documented in however.

Reply:
> I believe @robmoffat is of the opposite opinion?

Not really.  For me the slice point is:  does it travel between the DA and the proxy over the message port / channel?  If so, it's DACP, otherwise WCP.

(my reasoning behind this is that we abstract away the message port, and we could potentially use different media to pass these messages, e.g. sockets, web sockets, udp etc)

Reply:
Fair enough - I just thought they were a bit web-centric (iframe, CSS etc.), so less likely to apply to other languages. But as aI said, not that bothered.

Reply:
- If I'm understanding this correctly, would it be clearer to be explicit and say "Sent by the iframe to its `window.parent` frame via postMessage" (this style of explanation was used when describing WCP1Hello in WCP section).
- Far more importantly - and apologies if I'm misunderstanding - but is "iFrameHello" actually used as-is in the type properly from the message sent via postMessage API?  (It seems like it is if the version of packages/fdc3-get-agent/src/ui/AbstractUIComponent.ts that I looked at is the right one)
- If "iFrameHello" was actually used as-is for the type, this would seem to be a bad idea, as it feels far too generic for something posted to a window (as opposed to a specific, non-public channel via a MessagePort, where the subsequent comms take place) because other libraries used by the app (or even the app itself) could post messages with the same type.
- **Can we change this to using namespacing on the type e.g. something like "org.finos.fdc3.dacp.iFrameHello" ?**
- I appreciate specifying dacp or wcp in the namespace might be contentious given the previous comment/discussion between you guys.  I'm really just focused on using something like a "org.finos.fdc3." prefix tbh.
- Whatever string we use for the type here, we'll be stuck with it for a long time.  It would be really awful if we wanted to add namespaces to postMessage types sometime after v2.2 in order to disambiguate and prevent message type clashes, but then still had to make the npm package support non-namespaced message types for backward compatibility.
- (Feel free to ignore the following but) more trivially, I personally dislike the capital F used in 'iFrame'. I know it technically stands for Inline Frame, but it feels like people probably wouldn't use this style so freely if iPhone had never existed. I believe the HTML5 spec uses term term 'iframe' rather than 'iFrame' :)

Reply:
As discussed in a call with @novavi and others on this, we've agreed to change the name of the iframe messages to make clashes less likely.

That change is on a PR so it can be more easily checked before merging: https://github.com/finos/FDC3/pull/1375